app/controllers/restaurants_controller.rb

```diff
@@ -2,7 +2,15 @@ class RestaurantsController < ApplicationController
   before_action :set_restaurant, only: [:show, :update, :destroy]
 
   def index
-    render json: Restaurant.all
+    if request.query_parameters.present?
+      params = create_where_params_obj(request.query_parameters)
+
+      @restaurants = Restaurant.where(params)
```

Comment:
To avoid getting parameters you didn't mean to get we user `permit` and to make sure you get the parameters you need we use `require`. They're methods of the `Parameters` object Oded mentioned - https://api.rubyonrails.org/classes/ActionController/Parameters.html